bitnami/postgresql-ha/README.md

```diff
@@ -160,6 +160,12 @@ The following table lists the configurable parameters of the PostgreSQL HA chart
 | `postgresql.initdbScripts`                      | Dictionary of initdb scripts                                                                                                                                                                    | `nil`                                                        |
 | `postgresql.initdbScriptsCM`                    | ConfigMap with the initdb scripts (Note: Overrides `initdbScripts`). The value is evaluated as a template.                                                                                      | `nil`                                                        |
 | `postgresql.initdbScriptsSecret`                | Secret with initdb scripts that contain sensitive information (Note: can be used with initdbScriptsCM or initdbScripts). The value is evaluated as a template.                                  | `nil`                                                        |
+| `postgresql.tls.enabled`                        | Enable TLS traffic support for end-client connections                                                                                                                                           | `false`                                                      |
+| `postgresql.tls.preferServerCiphers`            | Whether to use the server's TLS cipher preferences rather than the client's                                                                                                                     | `true`                                                       |
+| `postgresql.tls.certificatesSecret`             | Name of an existing secret that contains the certificates                                                                                                                                       | `nil`                                                        |
+| `postgresql.tls.certFilename`                   | Certificate filename                                                                                                                                                                            | `""`                                                         |
+| `postgresql.tls.certKeyFilename`                | Certificate key filename                                                                                                                                                                        | `""`                                                         |
+| `postgresql.tls.certCAFilename`                 | CA Certificate filename. If provided, PgPool will authenticate TLS/SSL clients by requesting them a certificate.                                                                                | `nil`                                                        |
```

Comment:
Is Pgpool conscious of all this? It looks to me like you're only using these certificates in the PostgreSQL side

Reply:
Pgpool will connect to the backend using TLS once those parameters are enabled.
For example, this is the output when only enabling TLS in postgresql:

```
$ kubectl run mypg-postgresql-ha-client --rm --tty -i --restart='Never' --namespace default --image docker.io/bitnami/postgresql-repmgr:11.12.0-debian-10-r14 --env="PGPASSWORD=$POSTGRES_PASSWORD"  \
>         --command -- psql -h mypg-postgresql-ha-pgpool -p 5432 -U postgres -d postgres
If you don't see a command prompt, try pressing enter.

postgres=# 
postgres=# 
postgres=# pgpool show ssl;
 ssl 
-----
 off
(1 row)

postgres=# select * from pg_stat_ssl;
 pid | ssl | version |         cipher         | bits | compression | clientdn 
-----+-----+---------+------------------------+------+-------------+----------
 273 | f   |         |                        |      |             | 
 276 | f   |         |                        |      |             | 
 281 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 363 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 362 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 438 | f   |         |                        |      |             | 
 271 | f   |         |                        |      |             | 
 270 | f   |         |                        |      |             | 
 272 | f   |         |                        |      |             | 
(9 rows)
```

Reply:
Pgpool will connect to the backend using TLS once those parameters are enabled.
For example, this is the output when only enabling TLS in postgresql:

```
$ kubectl run mypg-postgresql-ha-client --rm --tty -i --restart='Never' --namespace default --image docker.io/bitnami/postgresql-repmgr:11.12.0-debian-10-r14 --env="PGPASSWORD=$POSTGRES_PASSWORD"  \
>         --command -- psql -h mypg-postgresql-ha-pgpool -p 5432 -U postgres -d postgres
If you don't see a command prompt, try pressing enter.

postgres=# 
postgres=# 
postgres=# pgpool show ssl;
 ssl 
-----
 off
(1 row)

postgres=# select * from pg_stat_ssl;
 pid | ssl | version |         cipher         | bits | compression | clientdn 
-----+-----+---------+------------------------+------+-------------+----------
 273 | f   |         |                        |      |             | 
 276 | f   |         |                        |      |             | 
 281 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 363 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 362 | t   | TLSv1.3 | TLS_AES_256_GCM_SHA384 |  256 | f           | 
 438 | f   |         |                        |      |             | 
 271 | f   |         |                        |      |             | 
 270 | f   |         |                        |      |             | 
 272 | f   |         |                        |      |             | 
(9 rows)
```

Reply:
All right, thanks!